src/fidesops/ops/service/strategy.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC
+from typing import Any, Dict, Generic, List, Optional, Type, TypeVar
+
+from pydantic import ValidationError
+
+from fidesops.ops.common_exceptions import NoSuchStrategyException
+from fidesops.ops.common_exceptions import ValidationError as FidesopsValidationError
+from fidesops.ops.schemas.saas.strategy_configuration import StrategyConfiguration
+
+logger = logging.getLogger(__name__)
+T = TypeVar("T", bound="Strategy")
+C = TypeVar("C", bound=StrategyConfiguration)
+
+
+def _find_strategy_subclass(
+    cls: Type[Strategy], strategy_name: str
+) -> Optional[Type[Strategy]]:
+    if hasattr(cls, "name") and cls.name == strategy_name:
+        return cls
+    for sub in cls.__subclasses__():
+        found = _find_strategy_subclass(sub, strategy_name)
+        if found is not None:
+            return found
+    return None
+
+
+def _find_all_strategy_subclasses(cls: Type[T]) -> List[Type[T]]:
+    subs = [cls] if hasattr(cls, "name") else []
```

Comment:
i don't love that this creates a bunch of single element lists upon recursion, i feel like it's probably giving unnecessary overhead. an alternative i'm thinking of is just returning the singleton element if there's only one, and then checking for the return type of the method call on recursion. it'd be a little more verbose, but i'm thinking it's probably worth it?

curious of other opinions and/or ideas.